src/main/java/apoc/nodes/Nodes.java

```diff
@@ -303,6 +299,39 @@ public boolean isDense(@Name("node") Node node) {
         }
     }
 
+    @Procedure(value = "apoc.remove.nodes.withlabels", mode = Mode.WRITE)
+    @Description("apoc.remove.nodes.withlabels(['Person','Movie'],-1) | Will remove all the nodes having one of the given labels, via DETACH DELETE. Use a negative limit for no limit in deletion. Same limit used for each label.")
+    public Stream<MapResult> removeNodesWithLabels(@Name("labels") List<String> labels, @Name("limit") Long limit) {
+        StringBuilder query = new StringBuilder();
+
+        Map<String,Object> results = new HashMap<String, Object>();
+        HashMap<String, Object> parameters = new HashMap<>();
+
+        for( String label: labels) {
+            if(label != null && !label.trim().equals("")) {
+                query.setLength(0);
+                query.append("MATCH (n:");
+                query.append(label);
+                query.append(")");
+                if (limit > 0) {
+                    query.append(" WITH n LIMIT ");
```

Comment:
I think stringbuilder is overkill here, just use string.format
and make sure to use backticks arround the label